### PR TITLE
Align java examples with scala ones

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Align java examples with scala ones, for the subworkflow related things.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
It is a bit confusing that `SubWorkflow` in Java and Scala carries different meaning for different purpose. So we align the Java one with Scala.

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit-examples/src/main/java/org/flyte/examples/SubWorkflow.java [hunk 1]
```diff
@@ -17,50 +17,28 @@
...
-import com.google.auto.value.AutoValue;
 import org.flyte.flytekit.SdkBindingData;
 import org.flyte.flytekit.SdkWorkflow;
 import org.flyte.flytekit.SdkWorkflowBuilder;
-import org.flyte.flytekit.jackson.Description;
 import org.flyte.flytekit.jackson.JacksonSdkType;
 
 @AutoService(SdkWorkflow.class)
-public class SubWorkflow extends SdkWorkflow<SubWorkflow.Input, SubWorkflow.Output> {
+public class SubWorkflow extends SdkWorkflow<WelcomeWorkflow.Input, WelcomeWorkflow.Output> {
```
Comment (honnix) [inline flytekit-examples/src/main/java/org/flyte/examples/SubWorkflow.java:26]: This becomes exactly the same as Scala.

File: flytekit-examples/src/main/java/org/flyte/examples/SumWorkflow.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import com.google.auto.service.AutoService;
+import com.google.auto.value.AutoValue;
+import org.flyte.flytekit.SdkBindingData;
+import org.flyte.flytekit.SdkWorkflow;
+import org.flyte.flytekit.SdkWorkflowBuilder;
+import org.flyte.flytekit.jackson.Description;
+import org.flyte.flytekit.jackson.JacksonSdkType;
+
+@AutoService(SdkWorkflow.class)
+public class SumWorkflow extends SdkWorkflow<SumWorkflow.Input, SumWorkflow.Output> {
```
Comment (honnix) [inline flytekit-examples/src/main/java/org/flyte/examples/SumWorkflow.java:28]: This was previously `SubWorkflow` in Java.

File: flytekit-examples/src/main/java/org/flyte/examples/UberWorkflow.java [hunk 3]
```diff
@@ -24,7 +24,7 @@
 import org.flyte.flytekit.jackson.JacksonSdkType;
 
 @AutoService(SdkWorkflow.class)
-public class UberWorkflow extends SdkWorkflow<UberWorkflow.Input, SubWorkflow.Output> {
+public class UberWorkflow extends SdkWorkflow<UberWorkflow.Input, SumWorkflow.Output> {
```
Comment (honnix) [inline flytekit-examples/src/main/java/org/flyte/examples/UberWorkflow.java:27]: `SubWorkflow` to `SumWorkflow` change.

File: flytekit-examples/src/main/java/org/flyte/examples/WelcomeWorkflow.java [hunk 4]
```diff
@@ -26,7 +26,7 @@
 
 /** Example workflow that takes a name and outputs a welcome message. */
 @AutoService(SdkWorkflow.class)
-public class WelcomeWorkflow extends SdkWorkflow<WelcomeWorkflow.Input, AddQuestionTask.Output> {
+public class WelcomeWorkflow extends SdkWorkflow<WelcomeWorkflow.Input, WelcomeWorkflow.Output> {
```
Comment (honnix) [inline flytekit-examples/src/main/java/org/flyte/examples/WelcomeWorkflow.java:29]: To align with Scala implementation.

**Review by pablocasares — APPROVED** [review]

**Pull request merged**